datalad/support/annexrepo.py

```diff
@@ -113,10 +113,20 @@ def __init__(self, path, url=None, runner=None,
           fresh git clone). Note that if `create=True`, then initialization
           would happen
         """
-        super(AnnexRepo, self).__init__(path, url, runner=runner,
-                                        create=create)
+        fix_it = False
+        try:
+            super(AnnexRepo, self).__init__(path, url, runner=runner,
+                                            create=create)
+        except GitCommandError as e:
+            if create and "Clone succeeded, but checkout failed." in str(e):
+                fix_it = True
+            else:
+                raise e
 
         self.always_commit = always_commit
+        if fix_it:
```

Comment:
GitCommandError is logged at that point already

2015-10-12 13:14 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/support/annexrepo.py
> <https://github.com/datalad/datalad/pull/259#discussion_r41746074>:
>
> >
> >          self.always_commit = always_commit
> > +        if fix_it:
>
> Worth at least info or even warning log
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/259/files#r41746074>.
>